modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/scheduler/XSKSchedulerManager.java

```diff
@@ -53,7 +54,9 @@ public static void scheduleJob(XSKJobDefinition jobDefinition) throws SchedulerE
           return;
         }
 
-        CronTrigger trigger = newTrigger().withIdentity(triggerKey).withSchedule(cronSchedule(jobDefinition.getCronExpression())).build();
+        CronTrigger trigger = newTrigger().withIdentity(triggerKey).startAt(new Date(jobDefinition.getStartAt().getTime()))
+            .endAt(new Date(jobDefinition.getEndAt().getTime()))
+            .withSchedule(cronSchedule(jobDefinition.getCronExpression()).withMisfireHandlingInstructionDoNothing()).build();
```

Comment:
What will happen, if the `startTime` and/or the `endTime` is not configured?

Reply:
We were only handling this case in the facade, we moved the check here as this method is called when the job synchronizer synchronizes automatically.